internal/controller/authenticator/jwtcachefiller/jwtcachefiller_test.go

```diff
@@ -144,66 +149,340 @@ func TestController(t *testing.T) {
 		require.NoError(t, err)
 	}))
 
-	goodIssuer := server.URL
+	badMuxInvalidJWKSURI := http.NewServeMux()
+	badOIDCIssuerServerInvalidJWKSURI := tlsserver.TLSTestServer(t, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		tlsserver.AssertTLS(t, r, ptls.Default)
+		badMuxInvalidJWKSURI.ServeHTTP(w, r)
+	}), tlsserver.RecordTLSHello)
+	badMuxInvalidJWKSURI.Handle("/.well-known/openid-configuration", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		_, err := fmt.Fprintf(w, `{"issuer": "%s", "jwks_uri": "%s"}`, badOIDCIssuerServerInvalidJWKSURI.URL, "https://.café   .com/café/café/café/coffee/jwks.json")
+		require.NoError(t, err)
+	}))
+
+	badMuxInvalidJWKSURIScheme := http.NewServeMux()
+	badOIDCIssuerServerInvalidJWKSURIScheme := tlsserver.TLSTestServer(t, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		tlsserver.AssertTLS(t, r, ptls.Default)
+		badMuxInvalidJWKSURIScheme.ServeHTTP(w, r)
+	}), tlsserver.RecordTLSHello)
+	badMuxInvalidJWKSURIScheme.Handle("/.well-known/openid-configuration", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		_, err := fmt.Fprintf(w, `{"issuer": "%s", "jwks_uri": "%s"}`, badOIDCIssuerServerInvalidJWKSURIScheme.URL, "http://.café.com/café/café/café/coffee/jwks.json")
+		require.NoError(t, err)
+	}))
+
+	goodIssuer := goodOIDCIssuerServer.URL
+	badIssuerInvalidJWKSURI := badOIDCIssuerServerInvalidJWKSURI.URL
+	badIssuerInvalidJWKSURIScheme := badOIDCIssuerServerInvalidJWKSURIScheme.URL
+	someOtherIssuer := "https://some-other-issuer.com" // placeholder only for tests that don't get far enough to make requests
+	someOtherLocalhostIssuer := "https://127.0.0.1:443/some-other-issuer"
+	someOtherLocalhostIssuerParsed, err := url.Parse(someOtherLocalhostIssuer)
+	require.NoError(t, err)
+
+	nowDoesntMatter := time.Date(1122, time.September, 33, 4, 55, 56, 778899, time.Local)
+	require.NoError(t, err)
```

Comment:
NIT: this `require.NoError()` is a relic, it can be removed, there is no error to check here.

Reply:
next push.